app/src/main/java/com/itsaky/androidide/services/GradleBuildService.java

```diff
@@ -391,11 +391,11 @@ public boolean isBuildInProgress() {
 
   public void startToolingServer(@Nullable OnServerStartListener listener) {
     if (toolingServerThread != null && toolingServerThread.isAlive()) {
-      throw new ToolingServerAlreadyStartedException();
+      // None Here
+    } else {
+      toolingServerThread = new Thread(new ToolingServerRunner(listener));
+      toolingServerThread.start();
     }

```

Comment:
Why the `if` block is empty? Can we not simply inverse the condition?

Reply:
> Why the `if` block is empty? Can we not simply inverse the condition?

Right, when doing this the following exception occurs when opening any project: 
```
AndroidIDE crash report
Manufacturer: Xiaomi
Device: M2102J20SG
App version: 2.1.2-beta (212)

 Stacktrace: 
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Thread.isAlive()' on a null object reference
	at com.itsaky.androidide.services.GradleBuildService.startToolingServer(GradleBuildService.java:393)
	at com.itsaky.androidide.EditorActivity$1.onServiceConnected(EditorActivity.java:190)
	at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:2096)
	at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:2129)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:210)
	at android.os.Looper.loop(Looper.java:299)
	at android.app.ActivityThread.main(ActivityThread.java:8250)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1045)
```

That's why I didn't do the inverse condition, so I used ``if (toolingServerThread == null)``, as the server hasn't started yet, the object will still be null, causing the condition to not work correctly, can you use ``try and catch`` but I don't know if it would be a good action.

Reply:
I told you to simply inverse (negate) the condition. This is one of the most basic stuffs you should know about. If you haven't, learn the basics about Boolean Algebra.

The condition was `toolingServerThread != null && toolingServerThread.isAlive()`. The condition once negated, becomes `toolingServerThread == null || !toolingServerThread.isAlive()`.

Reply:
Yes, I know Boolean algebra, what happened was a misunderstanding when I read your comment, I ended up understanding to "revert" the Else condition to If.